apm-sniffer/optional-plugins/mybatis-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mybatis/SqlSessionOperationInterceptor.java

```diff
@@ -0,0 +1,71 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.mybatis;
+
+import java.lang.reflect.Method;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+public class SqlSessionOperationInterceptor implements InstanceMethodsAroundInterceptor {
+
+    public static final String MYBATIS_ENTRY_METHOD_NAME = "mybatis_entry_method_name";
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                             MethodInterceptResult result) throws Throwable {
+        String operationName = objInst.getClass().getName() + "." + method.getName();
+        AbstractSpan lastSpan = null;
+        if (ContextManager.isActive()) {
+            lastSpan = ContextManager.activeSpan();
+        }
+        if (lastSpan == null || lastSpan.getComponentId() != ComponentsDefine.MYBATIS.getId()) {
+            AbstractSpan span = ContextManager.createLocalSpan(operationName);
+            span.setComponent(ComponentsDefine.MYBATIS);
+            if (allArguments != null && allArguments.length != 0) {
+                Tags.MYBATIS_MAPPER.set(span, (String) allArguments[0]);
+            }
+            ContextManager.getRuntimeContext().put(MYBATIS_ENTRY_METHOD_NAME, operationName);
+        }
+    }
+
+    @Override
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                              Object ret) throws Throwable {
+        String operationName = objInst.getClass().getName() + "." + method.getName();
+        if (String.valueOf(ContextManager.getRuntimeContext().get(MYBATIS_ENTRY_METHOD_NAME)).equals(operationName)) {
```

Comment:
avoid String.valueOf(null)

Reply:
![image](https://user-images.githubusercontent.com/28091237/116063823-fbe13b00-a6b7-11eb-82f4-b7dae5f01f47.png)
It doesn't seem to be a big problem.